v3/lint/registration.go

```diff
@@ -292,6 +301,63 @@ func (r *registryImpl) WriteJSON(w io.Writer) {
 	}
 }
 
+func (r *registryImpl) SetConfiguration(ctx Configuration) {
+	r.configuration = ctx
```

Comment:
nit: save `ctx` for `context.Context` just to avoid confusing readers that are used to that custom.
```suggestion
func (r *registryImpl) SetConfiguration(cfg Configuration) {
	r.configuration = cfg
```